Atomic/syscontainers.py

```diff
@@ -232,28 +249,32 @@ def _checkout_system_container(self, repo, name, img, deployment, upgrade, value
         if os.path.exists(destination):
             shutil.rmtree(destination)
 
-        os.makedirs(rootfs)
+        if self.args.remote:
+            os.makedirs(destination)
+        else:
+            os.makedirs(rootfs)
 
         rev = repo.resolve_rev(imagebranch, False)[1]
         manifest = self._image_manifest(repo, rev)
 
-        rootfs_fd = None
-        try:
-            rootfs_fd = os.open(rootfs, os.O_DIRECTORY)
-            if manifest is None:
-                self._checkout_layer(repo, rootfs_fd, rootfs, rev)
-            else:
-                layers = SystemContainers.get_layers_from_manifest(json.loads(manifest))
-                for layer in layers:
-                    rev_layer = repo.resolve_rev("%s%s" % (OSTREE_OCIIMAGE_PREFIX, layer.replace("sha256:", "")), False)[1]
-                    self._checkout_layer(repo, rootfs_fd, rootfs, rev_layer)
-            self._do_syncfs(rootfs, rootfs_fd)
-        finally:
-            if rootfs_fd:
-                os.close(rootfs_fd)
-
-        if extract_only:
-            return
+        if not self.args.remote:
+            rootfs_fd = None
+            try:
+                rootfs_fd = os.open(rootfs, os.O_DIRECTORY)
+                if manifest is None:
+                    self._checkout_layer(repo, rootfs_fd, rootfs, rev)
+                else:
+                    layers = SystemContainers.get_layers_from_manifest(json.loads(manifest))
+                    for layer in layers:
+                        rev_layer = repo.resolve_rev("%s%s" % (OSTREE_OCIIMAGE_PREFIX, layer.replace("sha256:", "")), False)[1]
+                        self._checkout_layer(repo, rootfs_fd, rootfs, rev_layer)
+                self._do_syncfs(rootfs, rootfs_fd)
+            finally:
+                if rootfs_fd:
+                    os.close(rootfs_fd)
+
+            if extract_only:
```

Comment:
the `extract_system_container()` function is used my mount so that the content can be inspected.

Remote containers should honor `extract_only` as well, let's just create a symlink for now, we can fix mount later.



Perhaps we must create a symlink to the actual rootfs?